tezos_interop/tezos_interop.re

```diff
@@ -506,24 +506,65 @@ module Run_contract = {
     };
   };
 };
-module Listen_transactions = {
-  open Tezos_micheline;
-  type michelson = Micheline.node(int, Michelson_v1_primitives.prim);
-  let michelson_of_yojson = json => {
-    // TODO: do this without serializing
-    let.ok json =
-      Yojson.Safe.to_string(json) |> Data_encoding.Json.from_string;
-    try(
-      Ok(
-        Micheline.root(
-          Data_encoding.Json.destruct(Pack.expr_encoding, json),
-        ),
-      )
-    ) {
-    | _ => Error("invalid json")
+
+type michelson =
+  Tezos_micheline.Micheline.node(int, Michelson_v1_primitives.prim);
+let michelson_of_yojson = json => {
+  // TODO: do this without serializing
+  let.ok json = Yojson.Safe.to_string(json) |> Data_encoding.Json.from_string;
+  try(
+    Ok(
+      Tezos_micheline.Micheline.root(
+        Data_encoding.Json.destruct(Pack.expr_encoding, json),
+      ),
+    )
+  ) {
+  | _ => Error("invalid json")
+  };
+};
+
+module Fetch_validators: {
+  let _run:
+    (~rpc_node: Uri.t, ~confirmation: int, ~contract_address: Address.t) =>
+    Lwt.t(result(michelson, string));
+} = {
+  [@deriving to_yojson]
+  type input = {
+    rpc_node: string,
+    confirmation: int,
+    contract_address: string,
+  };
+  [@deriving of_yojson]
+  type output = {
+    status: string,
+    storage: michelson,
```

Comment:
This is still not addressed, the output of this function should not be a michelson, the name of the module is `Fetch_validators`.

Reply:
Also the _run function is not using this type

Reply:
The name of the module is `Fetch_validators` but it's not fetching validators, it's fetching michelson. So the name of the module needs to be different or the implementation should be different.

The module has an output type, which is not used as output(actually I don't think it's even used at all). So the name of the type needs to change or it should be the output.